block-explorer/backend/src/tests/mod.rs

```diff
@@ -0,0 +1,365 @@
+use std::cmp::Ordering;
+use std::fmt::Display;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::time::Duration;
+
+use axum_test::TestServer;
+use demo_stf::genesis_config::{get_genesis_config, GenesisPaths};
+use demo_stf::App;
+use jsonrpsee::ws_client::WsClientBuilder;
+use serde_json::Value;
+use sov_db::ledger_db::{LedgerDB, SlotCommit};
+use sov_modules_stf_template::{SequencerOutcome, TxEffect};
+use sov_risc0_adapter::host::Risc0Verifier;
+use sov_rng_da_service::{RngDaService, RngDaSpec};
+use sov_rollup_interface::mocks::{MockAddress, MockBlock, MockBlockHeader, MockHash};
+use sov_rollup_interface::services::da::DaService;
+use sov_rollup_interface::stf::StateTransitionFunction;
+use testcontainers::clients::Cli;
+use testcontainers::Container;
+use testcontainers_modules::postgres::Postgres as PostgresImage;
+
+use crate::api_v0::default_pagination_size;
+use crate::db::Db;
+use crate::indexer::index_blocks;
+use crate::AppStateInner;
+
+type PostgresContainer<'a> = Container<'a, PostgresImage>;
+
+fn populate_ledger_db() -> LedgerDB {
+    std::env::set_var("TXNS_PER_BLOCK", "10");
+
+    let start_height: u64 = 0u64;
```

Comment:
Is this code adapted from a benchmark? If yes then maybe it is possible to extract it and use it here and in benchmarks?

Reply:
Yes, it's adapter from `examples/demo-rollup/benches/node`. I've deduplicated some code with PR https://github.com/Sovereign-Labs/sovereign-sdk/pull/1042 but there's some missing. I'll create a tracking issue if it's okay.

Reply:
Tracked in https://github.com/Sovereign-Labs/sovereign-sdk/issues/1065.